### PR TITLE
Feature/bbb 867 walking time

### PR DESCRIPTION
**Pull request by stephen-bealine**

### CODE REVIEW

File: src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeController.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  @ModelAttribute("walkingTimeOptions")
+  public RadioOptionsGroup walkingTimeOptions(
+      @ModelAttribute(JOURNEY_SESSION_KEY) Journey journey) {
+
+    if (!journey.isValidState(getStepDefinition())) {
+      return null;
+    }
+
+    List options = new ArrayList();
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeController.java]: Do you get a warning here in intellij about <RadioOption> type thing in the List ... or has java moved on?

File: src/main/resources/templates/walking/walking-time.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<head>
+    <title th:text="#{${JOURNEY.who} + 'walkingTime.title'} + ' - ' + #{common.title.suffix}"></title>
+</head>
+<body>
+
+<main layout:fragment="content">
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+
+            <div th:replace="~{ fragments/errors :: govuk-error-summary(${errorSummary}) }"/>
```
Comment (pauljroberts) [inline src/main/resources/templates/walking/walking-time.html]: I think this errors stuff has changed in One of Alis tickets.  Don't know if it's in develop yet (need to check on what I'm doing too).

File: src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import uk.gov.dft.bluebadge.webapp.citizen.StandaloneMvcTestViewResolver;
+import uk.gov.dft.bluebadge.webapp.citizen.client.applicationmanagement.model.WalkingLengthOfTimeCodeField;
+import uk.gov.dft.bluebadge.webapp.citizen.controllers.journey.RouteMaster;
+import uk.gov.dft.bluebadge.webapp.citizen.model.Journey;
+import uk.gov.dft.bluebadge.webapp.citizen.model.RadioOptionsGroup;
+import uk.gov.dft.bluebadge.webapp.citizen.model.form.ApplicantForm;
+import uk.gov.dft.bluebadge.webapp.citizen.model.form.ApplicantType;
+import uk.gov.dft.bluebadge.webapp.citizen.model.form.walking.WalkingTimeForm;
+
+public class WalkingTimeControllerTest {
```
Comment (pauljroberts) [inline src/test/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeControllerTest.java]: There's a ControllerTestFixture that does some of the legwork.  Have a look and use or not.

File: src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeController.java (hunk 1 above)
Comment (stephen-bealine) [reply src/main/java/uk/gov/dft/bluebadge/webapp/citizen/controllers/walking/WalkingTimeController.java]: Yeah it did. Have added the generic

**Pull request merged**